processing/src/main/java/org/apache/druid/query/aggregation/GroupingAggregatorFactory.java

```diff
@@ -0,0 +1,282 @@
+  {
+    return new LongConstantAggregator(value);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    return new LongConstantBufferAggregator(value);
+  }
+
+  @Override
+  public VectorAggregator factorizeVector(VectorColumnSelectorFactory selectorFactory)
+  {
+    return new LongConstantVectorAggregator(value);
+  }
+
+  @Override
+  public boolean canVectorize(ColumnInspector columnInspector)
+  {
+    return true;
+  }
+
+  /**
+   * Replace the param {@code keyDimensions} with the new set of key dimensions
+   */
+  public GroupingAggregatorFactory withKeyDimensions(Set<String> newKeyDimensions)
+  {
+    return new GroupingAggregatorFactory(name, groupings, newKeyDimensions);
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return VALUE_COMPARATOR;
+  }
+
+  @JsonProperty
+  public List<String> getGroupings()
+  {
+    return groupings;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  public long getValue()
+  {
+    return value;
+  }
+
+  @Nullable
+  @Override
+  public Object combine(@Nullable Object lhs, @Nullable Object rhs)
+  {
+    if (null == lhs) {
+      return rhs;
+    }
+    return lhs;
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new GroupingAggregatorFactory(name, groupings, keyDimensions);
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.singletonList(new GroupingAggregatorFactory(name, groupings, keyDimensions));
+  }
+
+  @Override
+  public Object deserialize(Object object)
+  {
+    return object;
+  }
+
+  @Nullable
+  @Override
+  public Object finalizeComputation(@Nullable Object object)
+  {
+    return object;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    // The aggregator doesn't need to read any fields.
+    return Collections.emptyList();
+  }
+
+  @Override
+  public ValueType getType()
+  {
+    return ValueType.LONG;
+  }
+
+  @Override
+  public ValueType getFinalizedType()
+  {
+    return ValueType.LONG;
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return Long.BYTES;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    CacheKeyBuilder keyBuilder = new CacheKeyBuilder(AggregatorUtil.GROUPING_CACHE_TYPE_ID)
+        .appendStrings(groupings);
+    if (null != keyDimensions) {
+      keyBuilder.appendStrings(keyDimensions);
+    }
+    return keyBuilder.build();
+  }
+
+  /**
+   * Gives the list of grouping dimensions, return a long value where each bit at position X in the returned value
+   * corresponds to the dimension in groupings at same position X. X is the position relative to the right end. if
+   * keyDimensions contain the grouping dimension at position X, the bit is set to 1 at position X, otherwise it is
+   * set to 0. An example adapted from Microsoft SQL documentation
```

Comment:
I think when other databases have differences in functionality so that there doesn't seem to be a consistent behavior, we typically follow the PostgreSQL behavior instead of others (https://github.com/apache/druid/pull/7950#discussion_r296916538, #9337, https://github.com/apache/druid/pull/9488#discussion_r391092673, #10006, etc), which seems to be the opposite of this where 0 means membership of the grouping set and 1 is not. 

I think we should be consistent with the PostgreSQL behavior here as well.

Reply:
I have seen that. Though it seems weird that 0 bit indicates that dimension is present, it seems it's not just postgres. I also checked that a commercial data warehouse follows the same logic as Postgres. will change this.

Reply:
I checked this further @clintropolis . Few things 
 - Mysql and postgres don't have the `grouping` functions that take more than one argument. 
 - Here is calcite documentation 
  ```
 * <p>Accepts 1 or more arguments.
 * Example: {@code GROUPING(deptno, gender)} returns
 * 3 if both deptno and gender are being grouped,
 * 2 if only deptno is being grouped,
 * 1 if only gender is being groped,
 * 0 if neither deptno nor gender are being grouped.
 *
  ```
I am now thinking that we should stick with how calcite intends it to be.  what do you think?

Reply:
MySQL and Postgres do seem to take more than one expressions. Check out the docs of [postgres](https://www.postgresql.org/docs/13/functions-aggregate.html) and [mysql](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_grouping). Their behaviour seems consistent across popular database systems such as oracle and sql server. One exception is that sql server supports both `grouping()` and `grouping_id()` which behaves in an opposite way to each other. [Calcite's behaviour is also consistent](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/fun/SqlGroupingFunction.java). Perhaps you were looking at an old version of the comment. I think Druid's behaviour should also match to most of other database systems.

Reply:
Good catch @jihoonson. The doc is fixed in the most recent version. I will change our impl as well.